Add logging to connect_vbms on non-200 (Close: #21)

paultag opened the pull request.

paultag commented:
Split from #29

alex commented on config/initializers/connect_vbms.rb:
```diff
@@ -2,4 +2,18 @@
+    when :request
+      if data[:response_code] != 200
+        Rails.logger.info "Serverside failure! Server response: " + data[:response_body]
+      end
```
@plusjeff I believe you were the one who asked for logging, is this the logging you wanted?

(Would it make more sense to include this logging in `handle_vbms_error`, the exception instance has the response body on it. Right now this would result in 2x the logging)

plusjeff replied:
@alex : Yeah - definitely would make more sense to include the logging in `handle_vbms_error` and avoid 2x the logging.

plusjeff commented:
@paultag: There's some feedback from @alex and I. Can you also sync this up with the latest from `master`?

plusjeff commented:
@paultag Anything else needed here other than a sync with master?

paultag commented:
Should be good, running tests & updating!

alex commented:
@paultag err, it doesn't look like the feedback was actioned upon?

paultag commented on app/controllers/web_controller.rb:
```diff
@@ -42,7 +42,6 @@ def handle_vbms_errors(&block)
     begin
       return block.call
     rescue VBMS::ClientError => e
-      Rails.logger.info e
```
@alex @plusjeff Just removed logging from here, since we should catch it down below. Exception might be if a `VBMS::ClientError` is raised with a non-500 HTTP status code. Thoughts?

alex replied:
I don't think there are any interesting cases where that occur.

alex commented on config/initializers/connect_vbms.rb:
```diff
@@ -2,4 +2,18 @@
+    case event
+    when :request
+      if data[:response_code] != 200
+        Rails.logger.info "Serverside failure! Server response: " + data[:response_body]
```
Can we log this data in a slightly more strucutred format, and also include the request type, and status code?

paultag commented on config/initializers/connect_vbms.rb:
```diff
@@ -2,4 +2,23 @@
+            data[:response_code].to_s +
+            " (" + data[:request].class.name + ") " +
+            data[:response_body],
+        )
```
Alright, anyone want to give this a shot? cc @plusjeff

alex replied:
Can you add `data[:request].class` as well?

paultag replied:
Sure, no problem. I'll just replace `.class.name` with `.class`

plusjeff replied:
Small syntax error:
```
Uncaught exception: /Users/vacomaherj/GitHub/caseflow/config/initializers/connect_vbms.rb:14: syntax error, unexpected ','
            data[:response_body],
```

plusjeff commented on config/initializers/connect_vbms.rb:
```diff
@@ -2,4 +2,23 @@
+
+$vbms = VBMS::Client.from_env_vars(
+  logger: CaseflowLogger.new,
+  env_name: ENV["CONNECT_VBMS_ENV"],
```
Might be a syntax error here too

plusjeff commented:
@paultag After the two follow-up comments are fixed, this is ready to go. Tests out fine locally.

paultag commented:
Fixed up. Was writing a bunch of go last night, added that comma before
pushing without thinking. Doh. Thanks.

On Tue, Oct 27, 2015 at 1:50 PM, Jeff Maher <notifications@github.com>
wrote:

> Assigned #31
> <https://github.com/department-of-veterans-affairs/caseflow/pull/31> to
> @paultag <https://github.com/paultag>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/department-of-veterans-affairs/caseflow/pull/31#event-447095710>
> .
>



-- 
:wq

alex commented on config/initializers/connect_vbms.rb:
```diff
@@ -2,4 +2,23 @@
+        Rails.logger.info (
+            "HTTP " +
+            data[:response_code].to_s +
+            " (" + data[:request].class + ") " +
```
Err, sorry should be `.class.name`, brain fart :-(

Also, we might want to use the Ruby `"#{data[:request].class.name}"` syntax, I think that's more readable than `+`. What do you think?

paultag replied:
Sounds good to me, I'll change that over!

paultag commented on config/initializers/connect_vbms.rb:
```diff
@@ -2,4 +2,15 @@
+    case event
+    when :request
+      if data[:response_code] != 200
+        Rails.logger.info "HTTP #{data[:response_code].to_s} (#{data[:request].class.name}) #{data[:response_body]}"
```
Changed to be a bit cleaner. First time using this syntax, might be worth a closer scan by eye! (seems to work, though!)

alex replied:
You can drop the `to_s`. Otherwise this looks ok to me.

Also, should this be a level above `info`?

Also, can we `s/HTTP/[VBMS HTTP Error]/` or similar?

paultag replied:
http://guides.rubyonrails.org/v4.1/debugging_rails_applications.html#log-levels

> The default Rails log level is info in production mode and debug in development and test mode.

Swaped to error

plusjeff commented:
@paultag Can you merge master in?

@KHarshawat Please review the text and design.

paultag commented:
@plusjeff master merged, commit rebased
